### DISCUSSION TITLE
Updating formula with bottles and a failed PR already

### DISCUSSION THREAD

**Discussion by mathomp4** [Homebrew/discussions#1169] [Everyday usage] [answered]
All, I would like to update the [`boxes` formula](https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/boxes.rb) to update to v2.0.0 (soon to be v2.1.0 which *really* is the version I want). 

I have done `brew bump-formula-pr` before, and I'm wondering if that's how I should do this...but this update has a few differences from what I've done before. 

First, `boxes` now really should have:
```ruby
  depends_on "libunistring"
  depends_on "pcre"
```
because technically it does (Though, at least on my install it doesn't need it but they might be satisfied by other formulae already.)

Locally I whipped up what I think the formula should be and tested it but I could only do it with:
```
brew install --build-from-source boxes
```
because the bottles got in the way. 

So, second, is there anything different I need to do with a formula with bottles? Or is that handled in CI on GitHub? 

Third, I naively tried `bump-formula-pr` and got:
```console
❯ brew bump-formula-pr boxes --version=2.0.0
Error: These pull requests may be duplicates:
boxes 2.0.0 https://github.com/Homebrew/homebrew-core/pull/71224
Duplicate PRs should not be opened. Use --force to override this error.
```
where the last PR failed. Maybe because it missed the `depends_on`? 

So what is the right way for me to be a good citizen and try to update this?

**Comment by Rylan12** [comment, marked as answer]
It's fine to go ahead and open a new PR. You probably want to just create the PR manually instead of using `brew bump-formula-pr`, though, to include the `depends_on` changes. `bump-formula-pr` is great for simple version updates, but for more complex PRs (such as changing dependencies) it's not intelligent enough.

Don't worry about the bottles. We build the bottles in our CI on GitHub when you make a PR so the reason installing doesn't work is because the bottles don't exist.

**Reply by mathomp4** [reply]
@Rylan12 

Sounds good. I'll give it a go! Apologies if you get some aborted PRs as I try this manually :)

**Reply by mathomp4** [reply]
@Rylan12 New wrinkle:
```console
❯ brew audit --strict boxes
boxes:
  * Formula boxes contains deprecated SPDX licenses: ["GPL-2.0"].
    You may need to add `-only` or `-or-later` for GNU licenses (e.g. `GPL`, `LGPL`, `AGPL`, `GFDL`).
    For a list of valid licenses check: https://spdx.org/licenses/
Error: 1 problem in 1 formula detected
```
As far as I can tell, the [boxes license](https://github.com/ascii-boxes/boxes/blob/master/LICENSE) is GPL-2.0. Does this mean the package can't be upgraded? Or do I need to consult with the developer?

**Reply by carlocab** [reply]
Usually, the source headers or the readme files indicate whether it's `-only` or `-or-later`. If it's still unclear, you can consult with the developer.

**Reply by mathomp4** [reply]
Okay. I'll ping him.

**Reply by Rylan12** [reply]
Just took a quick glance and it looks to me like `-only` since I don't see any `-or-later` indicators.
